### PR TITLE
include bootstrap admin in super-user group, ensure tokens file is correct on upgrades

### PR DESCRIPTION
**Pull request by liggitt**
Fixes https://github.com/kubernetes/kubernetes/issues/39532

Possible issues with cluster bring-up scripts:

- [x] known_tokens.csv and basic_auth.csv is not rewritten if the file already exists
  * new users (like the controller manager) are not available on upgrade
  * changed users (like the kubelet username change) are not reflected
  * group additions (like the addition of admin to the superuser group) don't take effect on upgrade
  * this PR updates the token and basicauth files line-by-line to preserve user additions, but also ensure new data is persisted
- [x] existing 1.5 clusters may depend on more permissive ABAC permissions (or customized ABAC policies). This PR adds an option to enable existing ABAC policy files for clusters that are upgrading

Moved follow-up issues to https://github.com/kubernetes/kubernetes/issues/39990

### CODE REVIEW

File: cluster/gce/container-linux/configure-helper.sh [hunk 1]
```diff
@@ -144,7 +144,7 @@ function create-master-auth {
   fi
   local -r known_tokens_csv="${auth_dir}/known_tokens.csv"
   if [[ ! -e "${known_tokens_csv}" ]]; then
-    echo "${KUBE_BEARER_TOKEN},admin,admin" > "${known_tokens_csv}"
+    echo "${KUBE_BEARER_TOKEN},admin,admin,system:masters" > "${known_tokens_csv}"
```
Comment (liggitt) [inline cluster/gce/container-linux/configure-helper.sh]: hmm... the file existence check above means these won't get re-written on upgrade

**Comment by liggitt** [comment]
cc @cjcullen @deads2k @kubernetes/sig-auth-misc

File: cluster/gce/container-linux/configure-helper.sh [hunk 2]
```diff
@@ -814,16 +828,22 @@ function start-kube-apiserver {
     webhook_authn_config_volume="{\"name\": \"webhookauthnconfigmount\",\"hostPath\": {\"path\": \"/etc/gcp_authn.config\"}},"
   fi
 
-  params+=" --authorization-mode=RBAC"
+  local authorization_mode="RBAC"
+  if [[ -n "${ABAC_AUTHZ_FILE:-}" && -e "${ABAC_AUTHZ_FILE}" ]]; then
```
Comment (deads2k) [inline cluster/gce/container-linux/configure-helper.sh]: Comment declaring intent and get @cjcullen to confirm its doing what you think and we'll still catch e2e failures.

File: cluster/addons/e2e-rbac-bindings/admin-binding.yaml [hunk 3]
```diff
@@ -1,16 +0,0 @@
...
-  labels:
-    kubernetes.io/cluster-service: "true"
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: view
-subjects:
-- apiVersion: rbac/v1alpha1
-  kind: User
-  name: admin
```
Comment (deads2k) [inline cluster/addons/e2e-rbac-bindings/admin-binding.yaml]: did you find where this guy was trying to a list pods or something?

Comment (liggitt) [inline cluster/addons/e2e-rbac-bindings/admin-binding.yaml]: several cluster scripts (`get-kube-local.sh`, `cluster/vagrant/util.sh`, etc) do `kubectl get pods` as a health check on startup.

**Comment by liggitt** [comment]
@k8s-bot cvm gce e2e test this

**Comment by liggitt** [comment]
@cjcullen this is ready for now, will open follow ups for remaining issues

**Comment by cjcullen** [comment]
/lgtm

We also have a certificate that "used to be" a super-user. On GCE, it has CN "kubecfg", on GKE it's "client". Maybe the ABAC flag on upgrades + the "wide-open RBAC" instructions make that not as bad, but it's probably worth tracking as a followup.

**Pull request merged**